semver.py

```diff
@@ -261,6 +257,13 @@ def _compare_by_keys(d1, d2):
     return rccmp
 
 
+def _ensure_is_comparable(other):
+    comparable_types = (VersionInfo, dict, tuple)
+    if not isinstance(other, comparable_types):
+        raise NotImplementedError("other type %r must be in %r"
```

Comment:
https://docs.python.org/3/library/constants.html#NotImplemented

You need to return NotImplemented here. Python itself will raise proper TypeError if objects aren't comparable.

Reply:
Thanks @kxepal 
I also noticed https://stackoverflow.com/questions/878943/why-return-notimplemented-instead-of-raising-notimplementederror